[Database] Build DatabaseGisInterface

robertsonbrinker opened the pull request:
Working on #71

robertsonbrinker committed:
Made a batabse folde for all

robertsonbrinker committed:
Update DatabaseGisManager.java

robertsonbrinker committed:
Update DatabaseGisManager.java

robertsonbrinker committed:
Create DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Create DatabaseGisManagerClass.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Update DatabaseGisManagerClass.java

robertsonbrinker committed:
Update DatabaseGisManager.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Fixed DGI test class a bit

robertsonbrinker committed:
Merge branch 'Build-DatabseGisInterfaceTests' of https://github.com/andrewdimmer/csi-3370-software-project into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
more fixes

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Generate SVG and PNG Images for PlantUML Diagrams

robertsonbrinker committed:
Update DatabasePersistentStorageClass.java

robertsonbrinker committed:
Merge branch 'Build-DatabseGisInterfaceTests' of https://github.com/andrewdimmer/csi-3370-software-project into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Fixing tests

robertsonbrinker committed:
Update DatabasePersistentStorageClass.java

robertsonbrinker committed:
style fixes

robertsonbrinker committed:
style fixes

robertsonbrinker committed:
Update .gitignore

robertsonbrinker committed:
Created GisManagerStub

robertsonbrinker committed:
Merge branch 'Build-DatabseGisInterfaceTests' of https://github.com/andrewdimmer/csi-3370-software-project into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Delete .classpath

robertsonbrinker committed:
Delete .project

robertsonbrinker committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
fixes

robertsonbrinker committed:
more

robertsonbrinker committed:
changed location of stub

robertsonbrinker committed:
removed unwanted files

robertsonbrinker committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Update LocationDataPointTests.java

robertsonbrinker committed:
Update MiddlewareCommLinkManager.java

robertsonbrinker committed:
Delete .classpath

robertsonbrinker committed:
Delete .project

robertsonbrinker committed:
The Production Class

robertsonbrinker committed:
fixes

robertsonbrinker committed:
Delete .project

robertsonbrinker committed:
deinitalized stuff

robertsonbrinker committed:
fixes

kylepoterek committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
overload constructor

robertsonbrinker committed:
Merge branch 'Build-DatabseGisInterfaceTests' of https://github.com/andrewdimmer/csi-3370-software-project into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
working on it

robertsonbrinker committed:
Merge branch 'Build-DatabseGisInterfaceTests' of https://github.com/andrewdimmer/csi-3370-software-project into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Update DatabaseGisManagerStub.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Class to implementation

robertsonbrinker committed:
Delete .classpath

robertsonbrinker committed:
Delete .project

robertsonbrinker committed:
moved stub to test folder

robertsonbrinker committed:
Merge branch 'Build-DatabseGisInterfaceTests' of https://github.com/andrewdimmer/csi-3370-software-project into Build-DatabseGisInterfaceTests

andrewdimmer commented on .gitignore:
```diff
@@ -28,4 +28,6 @@ hs_err_pid*
\ No newline at end of file
+*~
+**/#*#
+/bin1/
```
Can we remove this line? I'm not sure what it is coming from or what it corresponds to.

andrewdimmer commented on src/main/java/edu/oakland/helper/database/#PersistentStorageManager.java#:
```diff
@@ -0,0 +1,46 @@
+package edu.oakland.helper.database;
```
This is a temporary file that should be deleted.

andrewdimmer commented on src/test/java/edu/oakland/test/database/placeHolder.txt:
```diff
@@ -0,0 +1 @@
+Text
```
This is a temporary file that should be deleted.

andrewdimmer commented on src/test/java/edu/oakland/test/database/DatabaseGisInterfaceTest.java:
```diff
@@ -0,0 +1,58 @@
+  void currentSatGetsNextSat() {
+    DatabaseGisManagerStub dgms = new DatabaseGisManagerStub();
+    DatabaseGisInterfaceImplementation dgi = new DatabaseGisInterfaceImplementation(dgms);
+    String currentSat = "";
+    assertEquals("", dgi.receiveNextSatRequest(currentSat));
```
To actually verify that everything is being passed directly, can you place a test string there, and then reference `currentSat` as the expected output?

andrewdimmer commented on src/main/java/edu/oakland/production/database/DatabaseGisManagerStub.java:
```diff
@@ -0,0 +1,23 @@
+public class DatabaseGisManagerStub implements DatabaseGisManager {
+
+  public String passNextSatRequest(String currentSat) {
+    return "";
```
Please make sure that this passes out the `currentSat` value, otherwise you don't really learn anything from this test.

andrewdimmer commented on src/test/java/edu/oakland/test/database/DatabaseGisInterfaceTest.java:
```diff
@@ -0,0 +1,58 @@
+  void modeInIsModeOut() {
+    DatabaseGisManagerStub dgms = new DatabaseGisManagerStub();
+    DatabaseGisInterfaceImplementation dgi = new DatabaseGisInterfaceImplementation(dgms);
+    String n = "";
```
To actually verify that everything is being passed directly, can you place a test string here?

andrewdimmer commented on src/test/java/edu/oakland/test/database/DatabaseGisInterfaceTest.java:
```diff
@@ -0,0 +1,58 @@
+  @Test
+  @DisplayName("Checks that manager isn't null")
+  void databaseGisManagerNotNull() {
+    DatabaseGisManagerStub dgms = new DatabaseGisManagerStub();
+    String currentSat = "";
+    assertEquals("", dgms.passNextSatRequest(currentSat));
```
This doesn't actually check that the manager isn't null, as you're not actually testing the `DatabaseGisInterfaceImplementation` class. Can you model this test off of the similar test in `LocationDataPointTests` and `LocationDataPoint`?

andrewdimmer reviewed (changes requested):
Couple changes to verify that the tests actually pass, rather than we just get lucky. Can you please take another look at the commented sections? Thank you!

robertsonbrinker committed:
Update .gitignore

robertsonbrinker committed:
Delete #PersistentStorageManager.java#

robertsonbrinker committed:
Delete placeHolder.txt

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Update DatabaseGisManagerStub.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Update DatabaseGisManagerStub.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Update DatabaseGisInterfaceImplementation.java

robertsonbrinker committed:
Update DatabaseGisInterfaceImplementation.java

robertsonbrinker committed:
Update DatabaseGisInterfaceImplementation.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Update DatabaseGisInterfaceImplementation.java

robertsonbrinker committed:
Merge branch 'main' into Build-DatabseGisInterfaceTests

andrewdimmer commented on src/main/java/edu/oakland/production/database/DatabaseGisInterfaceImplementation.java:
```diff
@@ -0,0 +1,35 @@
+   *
+   * @param m an instance of the DatabaseGisManagerImplementation class.
+   */
+
```
Remove the empty line between the Javadoc and the method; some IDEs have issues with that.
```suggestion
```

andrewdimmer commented on src/main/java/edu/oakland/production/database/DatabaseGisInterfaceImplementation.java:
```diff
@@ -0,0 +1,35 @@
+
+public class DatabaseGisInterfaceImplementation {
+
+  DatabaseGisManager manager;
```
Add an extra line between the private attribute and the constructor.
```suggestion
  DatabaseGisManager manager;
  
```

andrewdimmer commented on src/main/java/edu/oakland/production/database/DatabaseGisManager.java:
```diff
@@ -6,8 +6,8 @@
   public void passStoreRequest(LocationDataPoint locationDataPoint);
   
-  public void passModeRequest(String mode);
+  public String passModeRequest(String mode);
```
Generally setter should be void return because you know what you set. Also, we want to see if we can avoid changing the public interface unnecessarily, as that could break other code from Middleware.
```suggestion
  public void passModeRequest(String mode);
```

andrewdimmer commented on src/main/java/edu/oakland/production/database/DatabaseGisManagerStub.java:
```diff
@@ -0,0 +1,23 @@
+package edu.oakland.production.database;
```
Can we go through and add a private variable and a getter to the stub to store the mode, so we don't need to change the public interface, but can still test what that the request works?

andrewdimmer commented on src/test/java/edu/oakland/test/database/DatabaseGisInterfaceTest.java:
```diff
@@ -0,0 +1,61 @@
+    ));
+    int i = 0;
+    dgi.receiveStoreRequest(ldp);
+    assertEquals(null, dgms.getLocationDataPoint(i));
```
This test should be failing! You're test is saying that `LocationDataPoint` coming out is always null, which is not true!

robertsonbrinker committed:
Update DatabaseGisManagerStub.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

robertsonbrinker committed:
Update DatabaseGisManagerStub.java

robertsonbrinker committed:
Update DatabaseGisManagerStub.java

robertsonbrinker committed:
Update DatabaseGisManagerStub.java

robertsonbrinker committed:
Merge branch 'Build-DatabseGisInterfaceTests' of https://github.com/andrewdimmer/csi-3370-software-project into Build-DatabseGisInterfaceTests

robertsonbrinker committed:
Update DatabaseGisInterfaceImplementation.java

robertsonbrinker committed:
Update DatabaseGisInterfaceTest.java

andrewdimmer committed:
[Database] Streamline DatabaseGisManagerStub Location Getter

The original design was built to test both a getter and a setter in the DatabaseGisInterface, but the getter is actually in the DatabaseCommInterface so the stub was simplified.

andrewdimmer committed:
[Database] Changed Mode Setters Back to Void Return

Was chanced from original spec to return a string for testing; adjusted the testing to return the production spec back to as was mapped out in the UML diagrams.

andrewdimmer reviewed (approved):
You're all set to merge!
